docs/JSON-RPC-API-using-jmwalletd.md

```diff
@@ -0,0 +1,284 @@
+* TODO: document the utxos return value format.
+
+#### `directsend`
+
+Make a bitcoin payment from the wallet, without coinjoin.
+
+* HTTP Request type: POST
+* Route: `/api/v1/wallet/<string:walletname>/taker/direct-send`
+* POST body contents: {"mixdepth": string, "amount_sats": string, "destination": string}
+* Returns:
+  - on success, {"walletname": string, "txinfo": json-string}
+
+##### Notes
+
+* `walletname` in the request URL will be used as the filename for the wallet created on disk in `datadir/wallets`.
+* `mixdepth` in POST body is a string representing the mixdepth/account ("0".."4").
+* `amount_sats` in POST body is a string containing the integer number of sats to pay.
+* `destination` in POST body is the bitcoin address to receive the payment.
+
+#### `docoinjoin`
+
+Get a new address for deposits.
+
+* HTTP Request type: POST
+* Route: `/api/v1/wallet/<string:walletname>/taker/coinjoin`
+* POST body contents: {"mixdepth": string, "amount": string, "counterparties": string, "destination": string}
+* Returns:
+  - on success, {"coinjoin_started": bool}
+
+##### Notes
+
+* `walletname` in the request URL will be used as the filename for the wallet created on disk in `datadir/wallets`.
+* This action only *starts* the coinjoin process, a successful return indicates start OK, not completion.
+* TODO: link to notifications of progress and completion of coinjoin.
+
+#### `session`
+
+Check the status and liveness of the session.
+
+* HTTP Request type: GET
+* Route: `/api/v1/session`
+* Returns:
+  - on success, {"session": bool, "maker_running": bool, "coinjoin_in_process": bool, "wallet_name": string}
+
+##### Notes
+
+* This method's primary goal is to give a client the assurance that the backend is actually running.
+* TODO: the other return values will probably be removed as websocket notifications do the job better.
+* `session` is true if and only if there is an active authentication and unlocked wallet.
+* `walletname` in response is the Joinmarket internal wallet name, `joinmarket-<hex encoded 3 byte hash identifier>`.
+
+
+#### `maker-start`
+
+Starts the yield generator/maker service for the given wallet, using the IRC and tor network connections
+in the backend (inside the process started with jmwalletd).
+See Joinmarket yield generator config defaults in `jmclient.configure` module for info on the data that must
+be specified in the POST body contents.
+
+* HTTP Request type: POST
+* Route: `/api/v1/wallet/<string:walletname>/maker/start`
+* POST body contents: {"txfee": string, "cjfee_a": string, "cjfee_r": string, "ordertype": string, "minsize": string}
+* Returns:
+  - on success, {"walletname": string}
+
+##### Notes
+
+* `walletname` in the request URL will be used as the filename for the wallet created on disk in `datadir/wallets`.
+* `walletname` in response is the Joinmarket internal wallet name, `joinmarket-<hex encoded 3 byte hash identifier>`.
```

Comment:
The `walletname` in the response content I get is equal to the requested wallet name, not the hash identifier, this also happens for all the other methods with supposedly such return value.  
Is this intentional?